driver-core/src/test/unit/com/mongodb/internal/connection/AbstractServerDiscoveryAndMonitoringTest.java

```diff
@@ -147,7 +147,7 @@ protected void init(final ServerListenerFactory serverListenerFactory, final Clu
         if (settings.getMode() == ClusterConnectionMode.SINGLE) {
             cluster = new SingleServerCluster(clusterId, clusterSettings, factory);
         } else {
-            cluster = new MultiServerCluster(clusterId, clusterSettings, factory);
+            cluster = new MultiServerCluster(clusterId, clusterSettings, factory, null /* TODO */);
```

Comment:
Maybe null is ok...

Reply:
Seems it is ok as long as `settings.getSrvHost() == null`. The MultiServerCluster class should defensively check that the resolver factory exists if there is a srv host.

Reply:
Handled by making this abstract and adding sub-classes, as you suggest below.